src/codegen/llvm/codegen_llvm.cc

```diff
@@ -1002,6 +1003,24 @@ llvm::Value* CodeGenLLVM::VisitExpr_(const Ramp* op) {
   return vec;
 }
 
+llvm::Value* CodeGenLLVM::VisitExpr_(const Shuffle* op) {
+  std::vector<llvm::Value *> vecs(op->vectors.size());
+  for (int i = 0; i < op->vectors.size(); ++i) {
+    vecs[i] = VisitExpr(op->vectors[i]);
+  }
+  llvm::Value* v0 = CreateVecConcat(vecs);
+  std::vector<uint32_t> idx(op->indices.size());
+  for (int i = 0; i < op->indices.size(); ++i) {
+    if (auto uimm = op->indices[i].as<UIntImm>()) {
+      idx[i] = uimm->value;
+    } else if (auto imm = op->indices[i].as<IntImm>()) {
+      idx[i] = imm->value;
+    }
+  }
+  llvm::Value* mask = llvm::ConstantDataVector::get(builder_->getContext(), idx);
```

Comment:
Does this mean we only have `unit32 x 8` for avx-512 and `uint x 4` for avx2? Is that possible that we have other types for `idx`?

Reply:
no. i believe we can put whatever we want in llvm ir, and llvm backend will handle this for us.